File: guides/micronaut-views-thymeleaf/java/src/main/java/example/micronaut/Photo.java
package example.micronaut;

import io.micronaut.core.annotation.ReflectiveAccess;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable // <1>
@ReflectiveAccess // <2>
public record Photo(Long albumId, String title, String url, String thumbnailUrl) {
}

File: guides/micronaut-views-thymeleaf/python/src/example/micronaut/photo.py
from dataclasses import dataclass

from micronaut.core.annotation import ReflectiveAccess
from micronaut.serde.annotation import Serdeable


@Serdeable  # <1>
@ReflectiveAccess  # <2>
@dataclass
class Photo:
    albumId: int
    title: str
    url: str
    thumbnailUrl: str
